### ISSUE TITLE
Profile Switch Error

### ISSUE THREAD

**Issue by ReneHezser** [mammuth/ha-fritzbox-tools#117]
I am using 1.11.2 on my FB 7490 running 7.20 and try to activate profile switch only but get an error in the logs:
```
2020-10-02 20:28:39 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
2020-10-02 20:29:09 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 617, in _async_registry_updated
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 320, in _async_write_ha_state
    sstate = self.state
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 667, in state
    return STATE_ON if self.is_on else STATE_OFF
  File "/config/custom_components/fritzbox_tools/switch.py", line 463, in is_on
    return self._is_on
AttributeError: 'FritzBoxProfileSwitch' object has no attribute '_is_on'
2020-10-02 20:29:39 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
```
The connectivity sensor is working:
![image](https://user-images.githubusercontent.com/5988767/94957466-166e7f80-04ee-11eb-9e86-8dcf84d8c51b.png)

The profile is associated to two devices (which is actually one with LAN and Wifi, but appears as two entries in the FB)

Is there anything else I can provide for troubleshooting?

**Comment by AaronDavidSchneider** [comment]
Thanks for the report! This seems to relate to a bug I found a couple days ago in the yaml Setup. Did this occur to you in yaml configuration or in the ui Setup?

**Comment by ReneHezser** [comment]
First I added thee config to the config.yaml. Then I read the recommended way was to use the UI, so I changed to it. I think I've restarted HA with with the config configuration before switching to UI config. But I can't say for certain.

**Comment by AaronDavidSchneider** [comment]
Ok thanks for the report. I think I have the bug fix included in PR #116.

**Comment by AaronDavidSchneider** [comment]
@ReneHezser, could you test PR #116? And tell if it fixes your bug?

**Comment by ReneHezser** [comment]
The error is gone. That's the good news. It still does not allow me to switch the profile though :-(
```
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] Setting up switches
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_wifi: False
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_profiles: True
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_deflections: False
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_port: False
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Setting up sensors
2020-10-05 12:37:34 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
2020-10-05 12:37:34 DEBUG (SyncWorker_18) [custom_components.fritzbox_tools.switch] Setting up profile switches
2020-10-05 12:38:34 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
```
From the integration screen:
![image](https://user-images.githubusercontent.com/5988767/95070220-15189f00-0708-11eb-86ae-02e5a2469515.png)
And the controls on lovelace
![image](https://user-images.githubusercontent.com/5988767/95070365-4abd8800-0708-11eb-9d5e-accc23ffefd9.png)

Any idea for troubleshooting?

**Comment by AaronDavidSchneider** [comment]
did you remove the integration and readded it (and restarted HA afterwards)?

**Comment by ReneHezser** [comment]
> did you remove the integration and readded it (and restarted HA afterwards)?

yes

**Comment by AaronDavidSchneider** [comment]
was this config.yaml or ui? Do you have more logs?

**Comment by ReneHezser** [comment]
> was this config.yaml or ui? Do you have more logs?

I configured it though the UI and do not have more logs. 
```custom_components.fritzbox_tools: debug``` is already present in my config to get the logs above.

**Comment by AaronDavidSchneider** [comment]
Thanks! The above log seems to be incomplete. Here is an example of my log.

```
2020-10-05 13:04:58 INFO (SyncWorker_62) [root] LOGGING IN TO FRITZ!BOX AT http://192.168.178.1...
2020-10-05 13:04:59 INFO (SyncWorker_62) [root] FETCHING THE PROFILE ID...
2020-10-05 13:05:00 INFO (MainThread) [homeassistant.setup] Setting up fritzbox_tools
2020-10-05 13:05:00 INFO (MainThread) [homeassistant.setup] Setup of domain fritzbox_tools took 0.0 seconds
2020-10-05 13:05:00 DEBUG (MainThread) [custom_components.fritzbox_tools] Setting up FRITZ!Box Tools component
2020-10-05 13:05:01 INFO (SyncWorker_14) [root] LOGGING IN TO FRITZ!BOX AT http://192.168.178.1...
2020-10-05 13:05:02 INFO (SyncWorker_14) [root] FETCHING THE PROFILE ID...
2020-10-05 13:05:03 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.fritzbox_tools
2020-10-05 13:05:03 INFO (SyncWorker_28) [homeassistant.loader] Loaded switch from homeassistant.components.switch
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Setting up sensors
2020-10-05 13:05:03 INFO (MainThread) [homeassistant.setup] Setting up switch
2020-10-05 13:05:03 INFO (MainThread) [homeassistant.setup] Setup of domain switch took 0.0 seconds
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
2020-10-05 13:05:03 INFO (MainThread) [homeassistant.components.switch] Setting up switch.fritzbox_tools
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] Setting up switches
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_wifi: False
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_profiles: True
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_deflections: False
2020-10-05 13:05:03 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_port: False
2020-10-05 13:05:03 DEBUG (SyncWorker_9) [custom_components.fritzbox_tools.switch] Setting up profile switches
2020-10-05 13:05:03 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new switch.fritzbox_tools entity: switch.fritzbox_profile_macbook
2020-10-05 13:05:03 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new binary_sensor.fritzbox_tools entity: binary_sensor.fritzbox_connectivity
```

**Comment by AaronDavidSchneider** [comment]
Did you find the missing lines @ReneHezser? Or maybe any other error?

**Comment by ReneHezser** [comment]
Unfortunately this is all I got from the log :-(
```
2020-10-05 12:37:30 DEBUG (MainThread) [custom_components.fritzbox_tools] Setting up FRITZ!Box Tools component

```
Is a line further up the log, but besides that there is no more info about fritzbox_tools.

**Comment by AaronDavidSchneider** [comment]
Ok. 

Thanks for the report. I have some more questions:
1. Do you have authentification via user and password enabled? (i.e. not via password only, see more in the readme)

the config process should look like this:

Discovery: Username and password that are also needed to login at `fritz.box`.
<img width="1188" alt="Screenshot 2020-10-05 at 15 21 36" src="https://user-images.githubusercontent.com/7556347/95084769-a1ce5780-071e-11eb-894a-8b1fd11e62e1.png">

Choose what you like to add:
<img width="497" alt="Screenshot 2020-10-05 at 15 22 05" src="https://user-images.githubusercontent.com/7556347/95084811-b01c7380-071e-11eb-957c-d5787db8cf0e.png">

Typing in profiles:
<img width="545" alt="Screenshot 2020-10-05 at 15 22 18" src="https://user-images.githubusercontent.com/7556347/95084831-b6aaeb00-071e-11eb-8556-1596ecf51862.png">

Wrong profile / profile not known in fritzbox:
<img width="623" alt="Screenshot 2020-10-05 at 15 22 37" src="https://user-images.githubusercontent.com/7556347/95084825-b6125480-071e-11eb-9be1-9face3ad22c0.png">

2. Could you try to add a false spelled profile and see if you get the above error?

**Comment by AaronDavidSchneider** [comment]
Regarding question 1:
<img width="1440" alt="Screenshot 2020-10-05 at 15 35 16" src="https://user-images.githubusercontent.com/7556347/95086166-751b3f80-0720-11eb-915f-048c1d804907.png">

**Comment by ReneHezser** [comment]
> 1. Do you have authentification via user and password enabled? (i.e. not via password only, see more in the readme)

Yes, I do have it enabled. Not using the username admin, but a custom name like in your screenshot. And the setting is activate in my FB as well.

Passing a wrong profile name returns *The format of the spcified profile(s) is wrong or profile not found in FRITZ!Box. Check your profiles in the FRITZ!Box at Internet/Filters/Access Profiles.*. So it seems the connection could be established (still nothing in the log).

Now with the correct profile name:
```
2020-10-05 16:06:35 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
2020-10-05 16:09:18 DEBUG (MainThread) [custom_components.fritzbox_tools] Setting up FRITZ!Box Tools component
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] Setting up switches
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_wifi: False
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_profiles: True
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_deflections: False
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.switch] use_port: False
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Setting up sensors
2020-10-05 16:09:22 DEBUG (MainThread) [custom_components.fritzbox_tools.binary_sensor] Updating Connectivity sensor...
2020-10-05 16:09:22 DEBUG (SyncWorker_34) [custom_components.fritzbox_tools.switch] Setting up profile switches
```
Restating HA does not help. The sensor still appears as unavailable.

**Comment by AaronDavidSchneider** [comment]
very strange... Any weird chars in the profile name? Or something like this? Have you tried to add a new profile in the FB and try the same again for a different profile?

**Comment by ReneHezser** [comment]
Just a name with with 4 chars (nothing special).

**Comment by AaronDavidSchneider** [comment]
Could you clone https://github.com/AaronDavidSchneider/fritzprofiles and execute the command
```
python fritzprofiles.py --user <user> --password <password> --url 192.168.178.1 --get_state --profile <profile_name>
```
in the directory `fritzprofiles/fritzprofiles`?

We should hopefully see the error there...

**Comment by ReneHezser** [comment]
> Could you clone https://github.com/AaronDavidSchneider/fritzprofiles and execute the command
> 
> ```
> python fritzprofiles.py --user <user> --password <password> --url 192.168.178.1 --get_state --profile <profile_name>
> ```
> 
> in the directory `fritzprofiles/fritzprofiles`?
> 
> We should hopefully see the error there...

There is no error. The result is "unlimited".

**Comment by AaronDavidSchneider** [comment]
Ok good to know. Just for completeness:
adding `--set_state unlimited` or `--set_state never` will toggle the state?!
If so we can conclude that the error is completely unrelated to the FB.

**Comment by ReneHezser** [comment]
Working as expected :-)

**Comment by AaronDavidSchneider** [comment]
Ok. I have absolutely no clue what else could be tried... I suppose that you already tried the following:

Remove the entry (integrations, fritzboxtools, remove). Remove the custom component completely (also the files). restart. add the files again. restart. discovery-> add entry.

Your error is not reproducable by me or other fritzbox_tools users. Your error is also not related to fritzbox connectivity issues. This means that there is something strange going on in home assistant :D :D

**Comment by ReneHezser** [comment]
> Remove the entry (integrations, fritzboxtools, remove). Remove the custom component completely (also the files). restart. add the files again. restart. discovery-> add entry.

I did it again :-)
No change, unfortunately.

**Comment by AaronDavidSchneider** [comment]
Solved in private communication. Fixed by #116.

**Comment by ReneHezser** [comment]
Thanks a lot
